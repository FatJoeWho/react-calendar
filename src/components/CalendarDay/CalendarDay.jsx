// @flow

import React from "react";
import styled from "styled-components";

const Div = styled.div`
	text-align: center;
	background: ${props => (props.selected ? "yellow" : "skyblue")};
	opacity: ${props => (props.fromOtherMonth ? "0.5" : "1")};
	grid-column: ${props => props.gridColumn};
	${props => props.styling};
`;

export default class CalendarDay extends React.Component {
	props: {
		displayContent: string,
		gridColumn: number
	};

	state = {
		displayContent: "day",
		gridColumn: 1
	};

	render() {
		return (
			<Div {...this.props}>
				{this.props.displayContent}
				{this.props.children}
			</Div>
		);
	}
}
