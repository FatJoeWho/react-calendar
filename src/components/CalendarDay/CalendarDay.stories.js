import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import CalendarDay from "components/CalendarDay";

storiesOf("Calendar Day", module)
	.add("Basic day", () => <CalendarDay displayContent={1} gridColumn={1} />)
	.add("Day with additional elements", () =>
		<CalendarDay displayContent={1} girdColumn={1}>
			<button
				onClick={e => {
					e.preventDefault;
					alert("Thank you");
				}}
			>
				Click me
			</button>
		</CalendarDay>
	)
	.add("Day with additional styling", () =>
		<CalendarDay
			displayContent={1}
			girdColumn={1}
			styling={"border: 20px solid green;"}
		/>
	);
