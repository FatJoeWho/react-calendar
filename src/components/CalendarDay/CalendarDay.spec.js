import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import CalendarDay from "components/CalendarDay";

describe("CalendarDay", () => {
	it("Should render", () => {
		const component = renderer.create(
			<CalendarDay gridColumn={1} displayContent={1} />
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Should allow child elements to be passed", () => {
		const sampleFunction = jest.fn();
		const component = mount(
			<CalendarDay gridColumn={1} displayContent={1}>
				<button
					onClick={e => {
						e.preventDefault;
						sampleFunction();
					}}
					className="btn"
				>
					Click me
				</button>
			</CalendarDay>
		);
		expect(component.find(".btn").length).toEqual(1);
		component.find(".btn").simulate("click");
		expect(sampleFunction.mock.calls.length).toBe(1);
	});
});
