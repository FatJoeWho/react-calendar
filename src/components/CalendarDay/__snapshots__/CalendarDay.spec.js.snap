// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay Should render 1`] = `
<div
  className="sc-bdVaJa fGkvKS"
>
  1
</div>
`;
